frontend-react/src/components/MessageTracker/MessageDetails.tsx

```diff
@@ -15,57 +18,154 @@ type MessageDetailsProps = {
     id: string | undefined;
 };
 
+const parseFileLocation = (
+    fileLocation: string
+): {
+    fileLocation: string;
+    sendingOrg: string;
+    fileName: string;
+} => {
+    let fileLocationFromStr = "";
+    let sendingOrgFromStr = "";
+    let fileNameFromStr = "";
+
+    let fileReportsLocation = fileLocation.split("reports/").pop();
+    if (fileLocation.includes("reports/") && fileReportsLocation) {
+        fileLocationFromStr = fileReportsLocation.split("%2F")[0].toUpperCase();
+        sendingOrgFromStr = fileReportsLocation.split("%2F")[1];
+        fileNameFromStr = fileReportsLocation.split("%2F")[2];
+    }
+
+    return {
+        fileLocation: fileLocationFromStr,
+        sendingOrg: sendingOrgFromStr,
+        fileName: fileNameFromStr,
+    };
+};
+
+const dataToAccordionItems = (props: {
+    id: string;
+    title: string;
+    data: WarningError[];
+}): AccordionItemProps[] => {
+    const results: AccordionItemProps[] = [];
+    if (props.data.length === 0) {
+        return [];
+    }
+    const items = (
+        <WarningsErrors title={""} data={props.data.map((md) => md.detail)} />
+    );
+    results.push({
+        id: props.id,
+        title: props.title,
+        expanded: false,
+        content: items,
+        headingLevel: "h3",
+        className: "",
+    });
+
+    return results;
+};
+
 export function MessageDetails() {
     const navigate = useNavigate();
     const { id } = useParams<MessageDetailsProps>();
     const { messageDetails } = useMessageDetails(id!!);
+    const warnings: WarningError[] = messageDetails
+        ? messageDetails.warnings
+        : [];
+    const errors: WarningError[] = messageDetails ? messageDetails.errors : [];
+
+    const fileUrl = messageDetails?.fileUrl ? messageDetails.fileUrl : "";
+    const { fileLocation, sendingOrg, fileName } = parseFileLocation(fileUrl);
 
     return (
         <>
             <div
                 className="grid-container margin-bottom-10"
                 data-testid="container"
             >
-                <div className="margin-bottom-5">
-                    <h1>Message Details</h1>
+                <div>
+                    <h2>Message ID Search</h2>
                     <Button onClick={() => navigate(-1)} type="button">
```

Comment:
In case a user got here directly / from a different page, is it worth actually setting the "Back" functionality to explicitly go to `/admin/message-tracker`?

Reply:
Yes, but we have a story coming to remove the back button and use breadcrumb instead.